core/src/main/scala/org/apache/spark/scheduler/cluster/ExecutorData.scala

```diff
@@ -35,5 +41,7 @@ private[cluster] class ExecutorData(
    var freeCores: Int,
    override val totalCores: Int,
    override val logUrlMap: Map[String, String],
-   override val attributes: Map[String, String]
-) extends ExecutorInfo(executorHost, totalCores, logUrlMap, attributes)
+   override val attributes: Map[String, String],
+   override val totalResources: Map[String, ResourceInformation],
+   val availableResources: mutable.Map[String, ExecutorResourceInfo]
```

Comment:
Does it need to be mutable? I don't find any operation that changes it in this PR.

Reply:
changed to immutable.Map